### DISCUSSION TITLE
How do the bindings internally works

### DISCUSSION THREAD

**Discussion by alkeryn** [wailsapp/wails#2499] [Q&A]
in the "how does it work" section the documentation explains how wails work by using bindings between go and js, but it does not explain how those bindings are made and work internally.

how is the communication between the go and javascript runtime achieved ?

thank you very much !

**Comment by leaanthony** [comment]
For v2, the app is built with a certain tag and run. This uses reflection to determine the methods bound and generates bindings in the form of typescript and JavaScript files. These are essentially wrappers around some internal JS methods that implement RPC with the Go backend. The backend, at startup, uses reflection to determine the signatures of the bound method and uses this after translating the call from JS to make the call in Go

**Reply by alkeryn** [reply]
hey thanks !
so how does the RPC work, is it simple http requests or a websocket to localhost ?

**Reply by leaanthony** [reply]
On v2, it's messages posted to the host from the webview. V3 will be internal http requests. Websockets is used when using browser in Dev mode.
